### PR TITLE
Adapt stream MTU on `connect`

### PR DESCRIPTION
**Pull request by kasperisager**

### CODE REVIEW

File: lib/raw-stream-set.js [hunk 1]
```diff
@@ -18,12 +18,29 @@ module.exports = class RawStreamSet {
...
+
+    async function onconnect () {
+      const {
+        remoteHost: host,
+        remotePort: port
+      } = stream
+
+      for (let mtu = 1400; mtu > 1200; mtu -= 100) {
+        try {
+          await self._dht.ping({ host, port }, { size: mtu - 12 /* overhead */ })
```
Comment (mafintosh) [inline lib/raw-stream-set.js:39]: pass the socket here also as options, stream.rawStream.socket so it has the correct session

Comment (kasperisager) [reply lib/raw-stream-set.js:39]: Terribly good point 😂

File: lib/raw-stream-set.js [hunk 2]
```diff
@@ -18,12 +20,35 @@ module.exports = class RawStreamSet {
...
+        try {
+          await self._dht.ping({ host, port }, {
+            size: mtu - 12, /* overhead */
+            socket: stream.socket
+          })
+
+          stream.setMTU(mtu)
+          break
+        } catch (err) {
+          safetyCatch(err)
```
Comment (mafintosh) [inline lib/raw-stream-set.js:49]: @kasperisager can you test what happens if this runs when the socket has been closed?

Comment (mafintosh) [reply lib/raw-stream-set.js:49]: ie, if the socket is opened and closed and then we run this ping in the background, just to make sure we don't crash becaue of that

Comment (kasperisager) [reply lib/raw-stream-set.js:49]: I'm not sure if I can construct a case where the socket is closed before the ping is attempted 🤔 The `catch` clause is hit during the other test cases though.

Comment (kasperisager) [reply lib/raw-stream-set.js:49]: @mafintosh Let me know if you have pointers on the above ☝️

**Comment by kasperisager** [comment]
Closing this in favour of MTU adaption directly in libudx.

**Pull request closed without merging**